gammapy/image/core.py

```diff
@@ -1088,3 +1091,82 @@ def assert_allclose(image1, image2, check_wcs=True):
             assert_wcs_allclose(image1.wcs, image2.wcs)
         else:
             raise ValueError('One image has `wcs==None` and the other does not.')
+
+    def convolve(self, kernel, fft=False, mode='reflect', **kwargs):
+        """
+        Convolve sky image with kernel.
+
+        Parameters
+        ----------
+        kernel : `~astropy.convolution.Kernel`
+            Convolution kernel.
+        fft : bool
+            Use fast FFT convolution.
+        mode : str ('reflect')
+            Boundary mode passed to `~numpy.pad`
+        **kwargs : dict
+            Further keyword arguments passed to `~numpy.pad`.
+        """
+        from scipy.signal import fftconvolve
+        from scipy.ndimage import convolve
+
+        width = kernel.array.shape[0] / 2
+        padded = self.pad(pad_width=(width, width))
+
+        if fft:
+            data = fftconvolve(padded.data, kernel.array, mode='same')
+        else:
+            data = convolve(padded.data, kernel.array, mode='constant')
+
+        convolved = self.__class__(data=data, wcs=self.wcs.copy())
+        return convolved.crop(crop_width=(width, width))
+
+    def __mul__(self, image):
+        """
+        Multiply sky image data.
+        """
+        # TODO: check if wcs are compatible, if not raise error
+        if self.wcs:
+            wcs = self.wcs.copy()
+        else:
+            wcs = None
+        data = self.data * image.data
+        return self.__class__(data=data, wcs=wcs)
+
+    def __truediv__(self, image):
+        """
+        Divide sky image data.
+        """
+        # TODO: check if wcs are compatible, if not raise error
+        if self.wcs:
+            wcs = self.wcs.copy()
+        else:
+            wcs = None
+        data = self.data / image.data
+        return self.__class__(data=data, wcs=wcs)
+
+
+class _SkyImageValidation(object):
+    """
+    Validate required input sky images for an algorithm.
+    """
+    def __init__(self, *args):
+        self.required_skyimages = args
+
+    @classmethod
+    def required(cls, *args):
+        return cls(*args)
+
+    def __call__(self, f):
+        @wraps(f)
+        def wrapped_f(*args):
+            for _ in self.required_skyimages:
+                if not _ in args[1].names:
+                    name = args[0].__class__.__name__
+                    raise ValueError("{} algorithm requires '{}' sky image"
+                                     " to run.".format(name, _))
+            return f(*args)
+        wrapped_f.__doc__ = f.__doc__.format(required=self.required_skyimages)
+        return wrapped_f
+
+required_skyimages = _SkyImageValidation.required
```

Comment:
`required_skyimage` should have a docstring.
Just put a string on the line after.

Also, how about exposing `required_skyimage` as `SkyImageList.check_required`?

Personally, I'd prefer the low-tech, easier to maintain and test code that just uses a method `SkyImageList.check_required` instead of implementing a decorator. It would just to this:
```
def check_required(self, names):
    if not set(names).issubset(self.names):
        raise ValueError(...)
```
and you'd call it like this on the first line of functions instead of applying the decorator:
```
images.check_required(['aaa', 'bbb'])
```